firehose_replicator/firehose_replicator.py

```diff
@@ -0,0 +1,53 @@
+import base64
+import boto3
+import json
+import os
+from datetime import datetime
+import uuid
+import gzip
+
+def lambda_handler(event, context):
+    output = []
+    records_for_s3 = []
+    stream_name = event['deliveryStreamArn'].split('/')[1]
+    target_bucket = os.environ.get('ECS_BUCKET_NAME')
+    key_prefix = os.environ.get('ECS_OBJECT_PREFIX')
+    full_key = "replicator_empty_key"
+
+    for record in event['records']:
+        print(f"Processing record {record['recordId']} ...")
+        file_content = base64.b64decode(record['data']).decode('utf-8')
+        timestamp_ns = record['approximateArrivalTimestamp']
+        timestamp_s = timestamp_ns / 1000
+        dt = datetime.fromtimestamp(timestamp_s)
+        ymdh_prefix = f"{dt.year:02}/{dt.month:02}/{dt.day:02}/{dt.hour:02}"
+        key_name = f"{stream_name}-{dt.year}-{dt.month:02}-{dt.day:02}-{dt.hour:02}-{dt.minute:02}-{dt.second:02}-{uuid.uuid1()}.gz"
+        full_key = f"{key_prefix}/{ymdh_prefix}/{key_name}"
```

Comment:
When this files are mirrored across accounts will they have the same filename? Or a different one? Note familiar with Firehose's naming standards...

Reply:
In short, **yes** the file name syntax will be the same. The above `full_key` is the syntax firehose uses, which is a timestamp + uuid.